core/trino-main/src/main/java/io/trino/operator/TrinoOperatorFactories.java

```diff
@@ -187,21 +196,39 @@ private OperatorFactory createJoinOperatorFactory(
                 .map(probeTypes::get)
                 .collect(toImmutableList());
 
-        return new LookupJoinOperatorFactory(
-                operatorId,
-                planNodeId,
-                lookupSourceFactoryManager,
-                probeTypes,
-                probeOutputChannelTypes,
-                lookupSourceFactoryManager.getBuildOutputTypes(),
-                joinType,
-                outputSingleMatch,
-                waitForBuild,
-                new JoinProbeFactory(probeOutputChannels.stream().mapToInt(i -> i).toArray(), probeJoinChannel, probeHashChannel),
-                blockTypeOperators,
-                totalOperatorsCount,
-                probeJoinChannel,
-                probeHashChannel,
-                partitioningSpillerFactory);
+        if (spillingEnabled) {
+            return new LookupJoinOperatorFactory(
+                    operatorId,
+                    planNodeId,
+                    (JoinBridgeManager<? extends LookupSourceFactory>) lookupSourceFactoryManager,
```

Comment:
You should create a new type hierarchy or new operator factory methods rather than explicitly cast here